src/fastoad/openmdao/tests/test_problem.py

```diff
@@ -148,19 +150,53 @@ def test_problem_read_inputs_with_nan_inputs(cleanup):
 
     problem.input_file_path = input_data_path
 
-    with pytest.raises(FASTOpenMDAONanInInputFile) as exc_info_1:
+    with pytest.raises(FASTOpenMDAONanInInputsError) as exc_info_1:
         problem.read_inputs()
     assert exc_info_1.value.input_file_path == input_data_path
     assert exc_info_1.value.nan_variable_names == ["x", "z"]
 
     problem.setup()
 
-    with pytest.raises(FASTOpenMDAONanInInputFile) as exc_info_2:
+    with pytest.raises(FASTOpenMDAONanInInputsError) as exc_info_2:
         problem.read_inputs()
     assert exc_info_2.value.input_file_path == input_data_path
     assert exc_info_2.value.nan_variable_names == ["x", "z"]
 
 
+def test_problem_read_inputs_with_missing_inputs(cleanup):
+    """
+    Tests that, when reading inputs using existing XML with missing value (for a
+    variable with default nan), an exception is raised.
+    """
+
+    problem = FASTOADProblem()
+    problem.model.add_subsystem(
+        "sellar",
+        SellarModel(
+            sellar_factory=GenericSellarFactory(
+                disc1_class=Disc1Quater, disc2_class=Disc2Bis, f_class=FunctionF
+            )
+        ),
+        promotes=["*"],
+    )
+
+    input_data_path = DATA_FOLDER_PATH / "missing_inputs.xml"
+
+    problem.input_file_path = input_data_path
+
+    with pytest.raises(FASTOpenMDAONanInInputsError) as exc_info_1:
+        problem.read_inputs()
+    assert exc_info_1.value.input_file_path == input_data_path
+    assert exc_info_1.value.nan_variable_names == ["z"]
+
+    problem.setup()
+
+    with pytest.raises(FASTOpenMDAONanInInputsError) as exc_info_2:
+        problem.read_inputs()
+    assert exc_info_2.value.input_file_path == input_data_path
+    assert exc_info_2.value.nan_variable_names == ["z"]
```

Comment:
Is the test before AND after the setup really needed ?

Reply:
Given that reading the entries does not involve exactly the same steps in these two cases, I feel more comfortable this way.